src/riak_core_bucket.erl

```diff
@@ -240,6 +241,17 @@ bucket_key({_Type, _Name}=Bucket) ->
 bucket_key(Name) ->
     {bucket, Name}.
 
+-spec fold_type(bucket_type(), fun((binary(), properties(), Acc) -> Acc), Acc) -> Acc.
```

Comment:
I had a bit of trouble wrapping my head around this mostly because I am still learning about the metadata system. Jason and I were thinking it might be more clear if the signature specified that the Acc is a proplist?

What do you think about:
```
-spec fold_type(bucket_type(), fun((binary(), properties(), Acc) -> Acc), Acc) ->
                       Acc when Acc :: [{atom(), term()}].
```

A doc comment might help here too.